README.md

```diff
@@ -34,6 +34,32 @@ The macro wraps a standard function to provide some extra functionality:
 * Whilst only providing what's specified in the _input schema_, the fn will only accept, as a return value, a map that validates against the _output schema_. Extraneous keys will be culled.
 * The return value is merged with the original input map, so that it will accrete over time.
 
+### defworkflowpred
+
+Use this macro to define a workflow predicate, used to control the flow of a loop (for example, using `do-while->`). They are structurally similar to workflow functions but only the truthiness of the return value is considered.
+
+```clojure
+(require '[witan.workspace-api :refer [defworkflowpred]]
+         '[schema.core :as s])
+
+(defworkflowpred my-predicate?
+  "This is a docstring" ;; optional
+  {:witan/name         :my-namespace/predicate-name
+   :witan/version      "1.0"
+   :witan/input-schema {:number s/Num}
+   :witan/param-schema {:value s/Num}}
```

Comment:
I've been thinking about this and I think a follow-up with examples would be appropriate but I'm reluctant to adjust this at the moment.